src/utils/src/Thread.c

```diff
@@ -198,6 +213,23 @@ PUBLIC_API STATUS defaultCreateThread(PTID pThreadId, startRoutine start, PVOID
     return retStatus;
 }
 
+PUBLIC_API STATUS defaultCreateThread(PTID pThreadId, startRoutine start, PVOID args)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+
+#if defined(KVS_DEFAULT_STACK_SIZE_BYTES)
+    CHK_STATUS(defaultCreateThreadWithParams(pThreadId, start, (SIZE_T) KVS_DEFAULT_STACK_SIZE_BYTES, args));
+#elif defined(CONSTRAINED_DEVICE)
+    CHK_STATUS(defaultCreateThreadWithParams(pThreadId, start, (SIZE_T) THREAD_STACK_SIZE_ON_CONSTRAINED_DEVICE, args));
```

Comment:
Although there is a check for the case of KVS_DEFAULT_STACK_SIZE_BYTES and CONSTRAINED_DEVICE both being defined through the CMakeLists, it is still possible to set both of these elsewhere such as #defines, in which case there is potential for hidden un-intended behavior. Can we consider adding a similar check within any of these if-defs?